docs/design/proposals/adoption/resource-adoption.md

```diff
@@ -0,0 +1,225 @@
+# Resource Adoption 
+
+# The Problem
+
+New ACK users typically already have implemented their infrastructure in existing solutions, meaning they have deployed production AWS resources into their accounts. While ACK provides good tooling for creating new resources, the path to migration from existing declarative infrastructure technologies into ACK is, so far, bumpy. The current method for migration requires users to store the current state of their applications, delete the resources, and re-create them using ACK. This process introduces numerous places for bugs, loss of the state of existing deployments and downtime in production workloads. Furthermore, considering the case where a K8s cluster loses state, users have no path to recover the previous resources without having to recreate all of them using ACK.
+
+ACK aims to provide the ability to “adopt” existing resources into your Kubernetes environment. Given the user has access to an existing resource, and that ACK operators exist for each resource’s associated service, the ACK controller will be able to determine the current specification and status of the AWS resource and reconcile said resource as if it were created natively in k8s.
+
+Users have requested this functionality as being pivotal to their adoption of ACK (see [#41](https://github.com/aws/aws-controllers-k8s/issues/41)). Information from user feedback has been considered as part of this document.
+
+Below are a number of use cases for requesting the ability this feature.
+
+1. Migrating existing architecture from another platform (CFN, CDK, Terraform, etc.)
+    1. Resources exist but have never been managed by K8s - no spec files
+2. Restoring from a broken K8s environment (etcd failure, complete state loss)
+    1. Resources exist and have previously been managed by K8s, but are not being actively reconciled
+3. Migrating architecture across K8s environments
+    1. Resource exist and are actively being managed by another K8s environment
```

Comment:
I wanted to call out that it could be any other k8s environment, because that is directly what the SageMaker team hopes to do from our current operator.

Reply:
Ah, gotcha. ++